pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go

```diff
@@ -62,7 +64,10 @@ type Cluster struct {
 // Validate calls each members Validate function
 func (c Cluster) Validate() error {
 	return validation.ValidateStruct(&c,
-		validation.Field(&c.ClusterRootDomain, validation.Required, is.LowerCase),
+		validation.Field(&c.ClusterRootDomain,
+			validation.Required,
+			is.LowerCase,
+			validation.NewStringRule(govalidator.IsDNSName, fmt.Sprintf("invalid domain name: '%s'", c.ClusterRootDomain))),
```

Comment:
I normally use `is.DNSName`. 

If you need to control the error message, you can use `.Error()` as in, `is.DNSName.Error("")`.

This way we don't have to directly depend on govalidator

Reply:
Fixed!